google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/internal/NameUtil.java

```diff
@@ -40,11 +45,73 @@ public static String formatTableName(
     return formatInstanceName(projectId, instanceId) + "/tables/" + tableId;
   }
 
+  public static String formatAuthorizedViewName(
+      @Nonnull String projectId,
+      @Nonnull String instanceId,
+      @Nonnull String tableId,
+      @Nonnull String authorizedViewId) {
+    return formatTableName(projectId, instanceId, tableId) + "/authorizedViews/" + authorizedViewId;
+  }
+
   public static String extractTableIdFromTableName(@Nonnull String fullTableName) {
     Matcher matcher = TABLE_PATTERN.matcher(fullTableName);
     if (!matcher.matches()) {
       throw new IllegalArgumentException("Invalid table name: " + fullTableName);
     }
     return matcher.group(3);
   }
+
+  public static String extractTableIdFromAuthorizedViewName(
+      @Nonnull String fullAuthorizedViewName) {
+    Matcher matcher = AUTHORIZED_VIEW_PATTERN.matcher(fullAuthorizedViewName);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException("Invalid authorized view name: " + fullAuthorizedViewName);
+    }
+    return matcher.group(3);
+  }
+
+  public static String extractTableNameFromAuthorizedViewName(
+      @Nonnull String fullAuthorizedViewName) {
+    Matcher matcher = AUTHORIZED_VIEW_PATTERN.matcher(fullAuthorizedViewName);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException("Invalid authorized view name: " + fullAuthorizedViewName);
+    }
+    return formatTableName(matcher.group(1), matcher.group(2), matcher.group(3));
+  }
+
+  public static String extractAuthorizedViewIdFromAuthorizedViewName(
+      @Nonnull String fullAuthorizedViewName) {
+    Matcher matcher = AUTHORIZED_VIEW_PATTERN.matcher(fullAuthorizedViewName);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException("Invalid authorized view name: " + fullAuthorizedViewName);
+    }
+    return matcher.group(4);
+  }
+
+  public static TargetId extractTargetId(
```

Comment:
Add a javadoc explaining this converts tableNamd and authorizedviewName to a TargetId